python/src/deltachat/tracker.py

```diff
@@ -90,11 +90,15 @@ def wait_progress(self, data1=None):
             if data1 is None or evdata == data1:
                 break
 
-    def wait_finish(self):
+    def wait_finish(self, timeout=None):
         """ wait until configure is completed.
 
-        Raise Exception if Configure failed
+        :param timeout: how many seconds until configuring is given up.
+        :raises: ConfigureFailed if Configured failed or timeout reached.
         """
-        if not self._configure_events.get():
-            content = "\n".join(map(str, self._ffi_events))
+        try:
+            if not self._configure_events.get(timeout=timeout):
+                content = "\n".join(map(str, self._ffi_events))
+                raise ConfigureFailed(content)
+        except Empty:
             raise ConfigureFailed(content)
```

Comment:
personally I had never had issues with accounts hanging forever in the configuration step they usually succeed or fail and end, but maybe there is some servers that actually cause the connection to hang, so it might be useful in some cases,

if you don't needed anymore then it if you don't want to waste your time with something you don't need, but you was really close to having it read just needed to add:

```suggestion
            content = "\n".join(map(str, self._ffi_events))
            raise ConfigureFailed(content)
```